src/legacy/core_plugins/kibana/public/dashboard/selectors/types.ts

```diff
@@ -47,8 +47,8 @@ export interface PanelState {
   readonly id: SavedObjectId;
   readonly version: string;
   readonly type: string;
-  readonly panelIndex: PanelId;
-  readonly embeddableConfig: any;
+  panelIndex: PanelId;
+  readonly embeddableConfig?: any;
```

Comment:
looks like `panelIndex` is no longer a `readonly` property?  is that necessary?
also is there a reason why `embeddableConfig` went from required to optional?

Reply:
When I made panelIndex readonly there's a problem in panel_utils `panel.panelIndex = (maxIndex++).toString();` Cannot assign to panelIndex because it is a read-only property. So I just removed the readonly part and it seems happy.

I had made embeddableConfig optional since __tests__ panel_state.createPanelWithDimensions doesn't return an embeddableConfig object. But adding an empty embeddableConfig option there makes more sense to me.  :) Jest tests run locally also seem cool with this - thanks!